arbnode/dataposter/data_poster.go

```diff
@@ -192,22 +192,17 @@ func (p *DataPoster[Meta]) getFeeAndTipCaps(ctx context.Context, lastTipCap *big
 
 	elapsed := time.Since(dataCreatedAt)
 	config := p.config()
-	maxFeeCap := new(big.Int).SetUint64(uint64(config.MaxFeeCapGwei * params.GWei))
-	maxFeeCapDoublings := int64(elapsed / config.MaxFeeCapDoubling)
-	// in tests, this could get way too big
-	if maxFeeCapDoublings > 8 {
-		maxFeeCapDoublings = 8
-	}
-	multiplier := new(big.Int).Exp(big.NewInt(2), big.NewInt(maxFeeCapDoublings), nil)
-	maxFeeCap.Mul(maxFeeCap, multiplier)
+	// MaxFeeCap = (BacklogOfBatches^2 * UrgencyGWei^2 + TargetPriceGWei) * GWei
+	maxFeeCap :=
+		new(big.Int).Mul(
+			new(big.Int).Add(
+				new(big.Int).Mul(
+					new(big.Int).SetUint64(backlogOfBatches*backlogOfBatches),
+					new(big.Int).SetUint64(uint64(config.UrgencyGwei*config.UrgencyGwei))),
+				new(big.Int).SetUint64(uint64(config.TargetPriceGwei))),
+			new(big.Int).SetUint64(uint64(params.GWei)))
```

Comment:
I think some of the `arbmath` package's helper functions could help simplify this and get rid of a lot of the `new(big.Int).`

Reply:
Done.